src/tribler-gui/tribler_gui/dialog_manager.py

```diff
@@ -0,0 +1,35 @@
+from typing import List, Optional, Type
+
+
+class DialogManager:
```

Comment:
I'd suggest doing it the way we did with TriblerRequestManager - create TriblerDialog class that will automatically register themselves with the singleton DialogManager.

Reply:
I've renamed `DialogContainer` to `TriblerDialog` (I'm not sure why it was named `DialogContainer` in the first place though). When creating a `TriblerDialog`, the constructor will automatically add this class to the `DialogManager`. When closing the dialog, it will unregister from the `DialogManager`. 👍